examples/lie_labs.py

```diff
@@ -0,0 +1,121 @@
+import torch
+
+import theseus.labs.lie as lie
+import theseus.labs.lie.functional as lieF
+
+batch_size = 5
+
+# ### Lie Tensor creation functions
+g1 = lie.rand(batch_size, lie.SE3, requires_grad=True)
+print(f"Created SE3 tensor with shape {g1.shape}")
+g2 = g1.clone()
+
+# Can create from a tensor as long as it's consistent with the desired ltype
+g3_data = lieF.so3.rand(5)
+g3 = lie.from_tensor(g3_data, lie.SO3)
+
+try:
+    x = lie.from_tensor(torch.zeros(1, 3, 3), lie.SO3)
+except ValueError as e:
+    print(f"ERROR: {e}")
+
+g4 = lie.as_lietensor(g3_data, lie.SO3)
+g5 = lie.cast(g3_data, lie.SO3)  # alias for as_lietensor
+
+# ### Lie operations
+v = torch.randn(batch_size, 6)
+
+# Exponential and logarithmic map
+out1 = g1.exp(v)  # also lie.exp(v, g1.ltype)
+print(f"Exp map returns a {type(out1)}.")
+out2 = g1.log()  # also lie.log(g1)
+print(f"Log map returns a {type(out2)}.")
+
+# Inverse
+out1 = g1.inv()  # also lie.inv(g1)
+
+# Compose
+# also lie.compose(g1, g2)
+out1 = g1.compose(g2)  # type: ignore
+
+# Differentiable jacobians
+jacs, out = g1.jcompose(g2)  # type: ignore
+print("Jacobians output is a 2-tuple.")
+print("    First element is a list of jacobians, one per group argument.")
+print(f"    For compose this means length {len(jacs)}.")
+print("    The second element of the tuple is the result of the operation itself.")
+print(f"    Which for compose is a {type(out).__name__}.")
+
+# Other options:
+#   * adj(), hat(), vee(), retract(), local(),
+#   * Jacobians: jlog(), jinv(), jexp()
+
+# ### Overriden operators
+# Local map
+z = g2 - g1  # equivalent to g1.local(g2)
```

Comment:
Another clarification question: `g1.local(g2)` is equivalent to `g2.inv().compose(g1)`, correct?
Is there a way to make these operations clear in terms of the order of operands?

Reply:
AFAIU, `g1.local(g2)` is `g1.inv().compose(g2)`. At least that's how we implemented in core Theseus. Any comments from @fantaosha.

Reply:
Hmm, so `g1.local(g2)` means "g2 in the local frame of g1". This is the exact type of operand order confusion I was mentioning. Or maybe it's just me that's confused.

Reply:
Actually `g2 - g1` seems like a perfectly good way to remove the confusion.

Reply:
I think @fantaosha had also mentioned some people use `+` for compose? I don't really have a strong preference here, except we should use what our typical user would be more accustomed to and/or find more natural.

Reply:
Here's what I, a single datapoint for a typical user, are accustomed to using:
`g1 * g2` for composing `g1` and `g2`
Explicitly doing `g1.inv() * g2` for locals, to remove any confusion as to what exactly is happening.

Reply:
Removed `__sub__` in 530172f5005a92b0eb52446092cd3f8620c673d9. We decided to only overload `*` for now, as seems to be the case in Sophus.